src/debugging/netSdk/NetSdkDebugHelper.ts

```diff
@@ -38,14 +46,70 @@ export class NetSdkDebugHelper extends NetCoreDebugHelper {
         return configurations;
     }
 
+    public async afterResolveDebugConfiguration(context: DockerDebugContext, debugConfiguration: DockerDebugConfiguration): Promise<void> {
+        const runTasks: Task[] = await tasks.fetchTasks({ type: NetSdkRunTaskType }) || [];
+        const runTaskFirstEntry = runTasks.find(t => t.source === NetSdkRunTaskType);
+        if (!runTaskFirstEntry) {
+            throw Error(l10n.t('No .NET SDK run task found '));
+        }
+        const taskExecution = await tasks.executeTask(runTaskFirstEntry);
+
+        const taskEndPromise = new Promise<void>((resolve, reject) => {
+            const disposable = tasks.onDidEndTask(e => {
+                if (e.execution === taskExecution) {
+                    disposable.dispose();
+                    resolve();
+                }
+            });
+        });
+
+        return taskEndPromise;
+    }
+
+    /**
+     * Overwite the base implementation to infer the container name from the project path (appProject)
+     * instead of the folder name
+     */
+    protected async loadExternalInfo(context: DockerDebugContext, debugConfiguration: DockerDebugConfiguration): Promise<{ configureSsl: boolean, containerName: string, platformOS: PlatformOS }> {
+        const associatedTask = context.runDefinition;
+        await ext.context.workspaceState.update(NetSdkProjPathMementoKey, debugConfiguration.netCore.appProject);
+
+        return {
+            configureSsl: !!(associatedTask?.netCore?.configureSsl),
+            containerName: this.inferDotNetSdkContainerName(debugConfiguration),
+            platformOS: associatedTask?.dockerRun?.os || 'Linux',
+        };
+    }
+
     /**
      * Checks if the launch task is using the .NET SDK Container build
      * @param preLaunchTask
      * @returns true if the launch task is using the .NET SDK Container build
      *          false otherwise
      */
-    public isDotNetSdkBuild(preLaunchTask: string): boolean {
-        return preLaunchTask === NetSdkTaskFullSymbol;
+    public isDotNetSdkBuild(platformConfiguration: unknown): boolean {
+        if (
+            typeof platformConfiguration === 'object' &&
+            platformConfiguration &&
+            'netCore' in platformConfiguration &&
+            (platformConfiguration as { netCore: { buildWithSdk: boolean } }).netCore.buildWithSdk
+        ) {
+            return true;
+        } else if (
+            typeof platformConfiguration === 'object' &&
+            platformConfiguration &&
+            'type' in platformConfiguration &&
+            (platformConfiguration as { type: string }).type === NetSdkRunTaskType
+        ) {
+            return true;
+        } else {
+            return false;
+        }
+    }
```

Comment:
Since you should only need this in the debug helper (see my comment about removing this check from the task platform check logic), you can simplify by just taking the NetCoreDebugOptions object.

```suggestion
    public isDotNetSdkBuild(netCoreDebugOptions?: NetCoreDebugOptions): boolean {
        return netCoreDebugOptions?.buildWithSdk ?? false;
    }
```

Reply:
Or just accept DockerDebugConfiguration and do `debugConfiguration.netCore?.buildWithSdk ?? false`